testsuite/tests/parsetree/source_jane_street.ml

```diff
@@ -444,6 +444,22 @@ let f ~(x1 @ many)
   stack_ (x1, x2, x3, x4, x5, x9, x10, x11, (* x12, *) x13, x14, x15)
 
 [%%expect{|
+val f :
+  x1:'b ->
+  x2:local_ string -> local_
```

Comment:
This is the part that looks wrong. It is printing `a -> b @ local -> (c -> d) @ local` which should always be equivalent to `a -> b @ local -> c -> d` and so should be printed as such.

Reply:
Actually, I see now that it is rather doing printing `a -> b @ local -> (c -> d) @ local unyielding`. (@riaqn We should really not print `local_` when we also need to print `@ foo` for some other `foo`, it is really confusing).

So I think the problem is rather the defaulting logic. It should be defaulting the yielding in the return mode to `yielding` since it cannot default the locality to `global`.

Reply:
> We should really not print local_ when we also need to print @ foo for some other foo, it is really confusing).

Yes #3522 already fixes that.